src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,294 @@
+			}
+			System.out.println("\n\nBase isnt HTTP\n\n");
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		System.out.println(newRequest.getRequestHeader().toString());
+		try{
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			System.out.println(newRequest); //Checking the request
+			String host = newRequest.getRequestHeader().getURI().getHost();
+			String path = newRequest.getRequestHeader().getURI().getPath();
+			newRequest.getRequestHeader().setURI(new URI("https",null, host, 443, path));
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		try {
+			String record = "";
+			do {
+				sendAndReceive(newRequest, false);
+				System.out.println("\n\nRequest Succesful");
+				String loc = "Location: ";
+				String response = newRequest.getResponseHeader().toString();
+				System.out.println(response);
+				int ind = response.indexOf(loc);
+				String found = response.substring(ind+10, response.length());
+				System.out.println(ind+" Index");
+				if(ind > 0) {
+					System.out.println(found +" The string");
+					String[] lines = found.split("\r\n|\r|\n");
+				    String newHost = lines[0].substring(0, lines[0].length()-1);
+				    System.out.println(newHost);
+					String path = newRequest.getRequestHeader().getURI().getPath();
+					if(newHost.contains("https")){
+						newHost = newHost.substring(8, newHost.length());
+					    newRequest.getRequestHeader().setURI(new URI("https", null, newHost, 443, path));
+					    record = "https";
+					} else if(newHost.contains("http")) {
+						newHost = newHost.substring(7, newHost.length());
+					    newRequest.getRequestHeader().setURI(new URI("http", null, newHost, 80, path));
+					    record = "http";
+					}
+					System.out.println(newRequest.getRequestHeader().toString());
+					System.out.println("\n\nThe record is "+record);	
+				}
+				else {
+					if(record.equals("http")) {
+						System.out.println("It was http");
+						String newUri = newRequest.getRequestHeader().getURI().toString();
+						bingo(getRisk(), //Risk
+								Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+								getName(), //Name
+								getDescription(), //Description
+								getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+								null, //Param
+								"", //Attack
+								Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", newUri), //OtherInfo 
+								getSolution(), //Solution
+								"", //Evidence
+								getCweId(), //CWE ID
+								getWascId(), //WASC ID
+								newRequest); //HTTPMessage
+					}
+					break;
+				}
+			}while(true);
+		} catch(NoRouteToHostException e) {
```

Comment:
Right, sorry, didn't notice that, SocketException is base class of NoRouteToHostException. The latter exception can be removed then.